### PR TITLE
Project Library Vittoria

### PR DESCRIPTION
**Pull request by vittoriamatteoli**
## Netlify link
https://thatcoollibrary.netlify.app/

## Collaborators
[vittoriamatteoli]

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  box-sizing: border-box;
+}
+
+.card-wrapper {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  background-color: #ffb000;
+  color: #b13f6a;
+  width: calc(33.33% - 40px); /* 33.33% for 3 columns, subtracting margins */
```
Comment (trista1987) [inline style.css:40]: `calc() `enable more flexibility and precision in layout 👍 clear explanation by leaving comment

File: index.html [hunk 2]
```diff
@@ -1,13 +1,54 @@
...
+    <title>That cool Library</title>
+  </head>
+
+  <body>
+    <h1>That cool Library</h1>
+    <section id="filterBar" class="filter-bar">
+      <div id="filtersOnGenre" class="filter-genre">
+        <h2>Filter by Genre</h2>
+        <button id="allBooks" class="filter-genre" onclick="showLibrary(books)">
+          All
```
Comment (trista1987) [inline index.html:22]: Button 'all' is a good idea for user click back to the main page

File: script.js [hunk 3]
```diff
@@ -4,403 +4,268 @@ of the properties make sense to filter on, and some to sort on.*/
...
     title: "Harry Potter and the Sorcerer's Stone",
-    author: 'J.K. Rowling',
+    author: "J.K. Rowling",
     year: 1997,
-    genre: 'Fantasy',
+    genre: "Fantasy",
     rating: 4.7,
     description:
-      'The first book in the beloved Harry Potter series, it introduces readers to the magical world of Hogwarts and the young wizard Harry Potter.',
-    image: "./books-images/harry-potter-and-the-sorcerer'.jpg"
```
Comment (trista1987) [inline script.js:74]: "Harry Potter and the Sorcerer's Stone" -- this img doesn't work on web page

File: index.html [hunk 4]
```diff
@@ -1,13 +1,54 @@
...
+        </button>
+        <button id="filterFiction" onclick="fictionSelect(books)">
+          Fiction
```
Comment (trista1987) [inline index.html:25]: I think if we've already used `addEventListener` on JS, we don't need to add `onclick` on html again?

**Review by trista1987 — COMMENTED** [review]
Code are clean, easy to read and comments well explained. Buttons work well. Web page displays all books. Only one picture doesn't work. Random button works. I like the 'all' button idea.

File: style.css [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  margin: 20px;
+  padding: 20px;
+  box-sizing: border-box;
+  border-radius: 20px;
+  border: solid 2px rgb(205, 117, 30);
+}
+
+button {
```
Comment (NellaMori) [inline style.css:48]: to improve the UI on mobile screens, reduce the size of paddings/margins for this element, it takes up too much space. Font size can be reduces too in the parent elements 👍 
Screenshot 2024-03-04 at 09 27 29

File: script.js [hunk 6]
```diff
@@ -4,403 +4,289 @@ of the properties make sense to filter on, and some to sort on.*/
...
+  const filteredMistery = books.filter((book) => book.genre === "Mystery");
+  showLibrary(filteredMistery);
+};
+filterMistery.addEventListener("click", misterySelect(books));
+
+const dystopianSelect = (books) => {
+  const filteredDystopian = books.filter((book) => book.genre === "Dystopian");
+  showLibrary(filteredDystopian);
+};
+filterDystopian.addEventListener("click", dystopianSelect(books));
```
Comment (NellaMori) [inline script.js:259]: this code is correct, but very verbose. Here a refactored suggestion:

```
const genreSelect = (genre) => {
  return () => {
    const filteredBooks = books.filter((book) => book.genre === genre);
    showLibrary(filteredBooks);
  };
};

filterFiction.addEventListener("click", genreSelect("Fiction"));
```

**Review by NellaMori — APPROVED** [review]
Hi Vittoria, congrats you did a really good job with this project 🔥 You followed the instructions and the app fulfils the requirements of filtering and responsiveness. Overall a very good execution considering you also worked on your own 👍 I left some comments around the code for little improvements, keep up the fantastic work! 🏅

**Pull request closed without merging**